agent/taskresource/fsxwindowsfileserver/fsxwindowsfileserver_windows.go

```diff
@@ -479,15 +478,24 @@ func (fv *FSxWindowsFileServerResource) retrieveSSMCredentials(credentialsParame
 	}
 
 	ssmClient := fv.ssmClientCreator.NewSSMClient(fv.region, iamCredentials)
-	ssmParam := filepath.Base(parsedARN.Resource)
-	ssmParams := []string{ssmParam}
+	// parsedARN.Resource looks like "arn:aws:ssm:us-west-2:123456789012:parameter/sample1/sample2/parameter1"
+	// We cut by parameter and get "arn:aws:ssm:us-west-2:123456789012:parameter", "/sample1/sample2/parameter1", True/False
+	_, ssmParamName, found := strings.Cut(parsedARN.Resource, "parameter")
+	if !found {
+		err = errors.New("unxpected error. expected fsx credential ssm arn but did not find string 'parameter' in the arn")
```

Comment:
How about adding `parsedARN.Resource` to the error string?

Reply:
I was hesitant for security reasons, there were some concerns around doing that when we implemented domainless gMSA

Reply:
oh okay